addon/templates/components/polaris-pagination.hbs

```diff
@@ -17,3 +19,17 @@
 }}
   {{polaris-icon source="arrow-right"}}
 {{/polaris-button/button}}
+
+{{#if isPreviousKeyListenerEnabled}}
+  {{key-press-listener
+    key=previousKeys
+    onKeyUp=(action onPrevious)
+  }}
+{{/if}}
```

Comment:
In essence, removing the component does the same w/o the extra logic to handle removal of listeners which we would have to add if we used a flag instead since its baked into the component's lifecycle events. 

I usually use flags like the one you mentioned when dealing with presentational components where we can avoid a reflow of DOM by passing in a flag & using that to hide parts of the dom with css instead of removing and adding the element from the dom. 

If you feel that there's a benefit I'm missing by not adding the prop to this component, please let me know 🙂

Reply:
The flag would prevent tearing down and re-initialising the component. Initially I was also thinking that it'd clean up the possible use case when the component is passed into another component; I guess you could put the logic in a number of places (outside the component as you have it currently, as a flag passed into the component, inside the action passed into the component etc.). To me it just seemed a bit cleaner to encapsulate that logic within the listener, but I'm OK with it staying like it is for now if that change needs a lot of work 👍